routes/postRoutes.js

```diff
@@ -0,0 +1,150 @@
+const express = require("express");
+const router = express.Router();
+
+const Db = require("../data/db");
+
+router.get("/", (req, res) => {
+  Db.find()
+    .then(posts => {
+      res.status(200).json(posts);
+    })
+    .catch(error => {
+      res
+        .status(500)
+        .json({ error: "The posts information could not be retrieved." });
+    });
+});
+
+router.get("/:id", (req, res) => {
+  const id = req.params.id;
+  Db.findById(id)
+    .then(post => {
+      if (post) {
```

Comment:
In this case, the way the db helpers were written, findById returns an array with the single object you were looking for. So in this case, when it doesn't find an entry at that id, it returns an empty object, which evaluates to truthy. So we should check the length of post instead 
```suggestion
      if (post.length > 0) {
```